From 3ebaa40606b5d13c765841eff31ff56792b64fe3 Mon Sep 17 00:00:00 2001
From: Jocelyn Falempe <jfalempe@redhat.com>
Date: Thu, 29 Jan 2026 09:51:11 +0100
Subject: [PATCH] agetty: Add -8 arguments to accept full 8-bits characters

kmscon sends utf-8 characters, which get silently dropped by agetty.
This leads to some weird behavior with accented letters.

Signed-off-by: Jocelyn Falempe <jfalempe@redhat.com>
---
 scripts/systemd/kmsconvt@.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/systemd/kmsconvt@.service b/scripts/systemd/kmsconvt@.service
index 946c92a2..40de29b7 100644
--- a/scripts/systemd/kmsconvt@.service.in
+++ b/scripts/systemd/kmsconvt@.service.in
@@ -38,7 +38,7 @@ IgnoreOnIsolate=yes
 ConditionPathExists=/dev/tty0
 
 [Service]
-ExecStart=@bindir@/kmscon --vt=%I --seats=seat0 --no-switchvt --login -- /sbin/agetty -o '-p -- \\u' --noclear -- - $$TERM
+ExecStart=@bindir@/kmscon --vt=%I --seats=seat0 --no-switchvt --login -- /sbin/agetty -8 -o '-p -- \\u' --noclear -- - $$TERM
 UtmpIdentifier=%I
 TTYPath=/dev/%I
 TTYReset=yes
